addons/mail/models/ir_attachment.py

```diff
@@ -34,3 +34,15 @@ def register_as_main_attachment(self, force=True):
                     related_record.message_main_attachment_id = self
                 except AccessError:
                     pass
+
+    def _attachment_format(self, main_attachment = None, safari = None):
```

Comment:
As already commented by MAT, `safari` should not be a param of this method, it should be computed inside it if we need it for a technical reason.

Also `main_attachment` does not make sense as a param here, this is contextual information to a message/thread, it shouldn't be in a generic attachment format method, but stay as logic in message.